reconcile/utils/saasherder.py

```diff
@@ -1038,8 +1130,19 @@ def init_populate_desired_state_specs(self, saas_file):
                 namespace = target["namespace"]["name"]
                 env_name = target["namespace"]["environment"]["name"]
 
-                key = f"{saas_file_name}/{rt_name}/{cluster}/" f"{namespace}/{env_name}"
-                digest = SaasHerder.get_target_config_hash(target_configs[key])
+                dummy_job_spec = TriggerSpecConfig(
+                    saas_file_name=saas_file_name,
+                    env_name=env_name,
+                    timeout=None,
+                    pipelines_provider=None,
+                    resource_template_name=rt_name,
+                    cluster_name=cluster,
+                    namespace_name=namespace,
+                    state_content=None,
+                )
+                digest = SaasHerder.get_target_config_hash(
+                    target_configs[dummy_job_spec.state_key]
+                )
```

Comment:
another nit: if this dumm job spec has no other purpose than creating the state_key, i would prefer to use it without assigning it. less mental state

```suggestion
                state_key = TriggerSpecConfig(
                    saas_file_name=saas_file_name,
                    env_name=env_name,
                    timeout=None,
                    pipelines_provider=None,
                    resource_template_name=rt_name,
                    cluster_name=cluster,
                    namespace_name=namespace,
                    state_content=None,
                ).state_key
                digest = SaasHerder.get_target_config_hash(
                    target_configs[state_key]
                )
```

Reply:
great idea!
https://github.com/app-sre/qontract-reconcile/pull/2782/commits/34a08a219b3ab8a5251b0caa9a4c7f4a35750c34